scripts/pages.mjs

```diff
@@ -43,6 +43,69 @@ class ClassPages extends Application {
     html[0].querySelectorAll(".class-nav img[data-action]").forEach(n => {
       n.addEventListener("click", this._onClickClass.bind(this));
     });
+    this.navScroller(html);
+  }
+
+  /**
+   * Activates the scroll functionality for the class icons gallery.
+   * @param {JQuery} html The HTML content of the rendered ClassPages application.
+   */
+  navScroller(html) {
+    const gallery = html[0].querySelector(".class-nav .gallery");
+    let isScrolling = false;
+
+    gallery.addEventListener(
+      "wheel",
+      function (event) {
+        if (isScrolling) return;
+
+        isScrolling = true;
+        const direction = Math.sign(event.deltaY);
+
+        if (direction > 0) {
+          this.scrollToNextIcon(gallery);
+        } else if (direction < 0) {
+          this.scrollToPreviousIcon(gallery);
+        }
+
+        setTimeout(() => {
+          isScrolling = false;
+        }, 100);
+      }.bind(this)
+    );
+  }
+
+  /**
+   * Scrolls to and activates the next icon in the gallery. If the current icon is the last one,
+   * it loops back to the first icon.
+   * @param {HTMLElement} gallery The gallery element containing the class icons.
+   */
+  scrollToNextIcon(gallery) {
+    const activeIcon = gallery.querySelector("img.active");
+    let nextIcon =
+      activeIcon.nextElementSibling || gallery.querySelector("img:first-child");
+
+    if (nextIcon) {
+      nextIcon.scrollIntoView({ behavior: "smooth", inline: "center" });
+      this._onClickClass({ currentTarget: nextIcon });
+    }
+  }
+
+  /**
+   * Scrolls to and activates the previous icon in the gallery. If the current icon is the first one,
+   * it loops back to the last icon.
+   * @param {HTMLElement} gallery The gallery element containing the class icons.
+   */
+  scrollToPreviousIcon(gallery) {
+    const activeIcon = gallery.querySelector("img.active");
+    let previousIcon =
+      activeIcon.previousElementSibling ||
+      gallery.querySelector("img:last-child");
```

Comment:
Definitely overkill with the linebreaks. We allow a maximum length of 125 characters.